src/components/pin/pin.repository.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+import { node, relation } from 'cypher-query-builder';
+import { DateTime } from 'luxon';
+import { Session } from '../../common';
+import { DatabaseService, matchRequestingUser } from '../../core';
+
+@Injectable()
+export class PinRepository {
+  constructor(private readonly db: DatabaseService) {}
+
+  async isPinned(id: string, session: Session): Promise<boolean> {
+    const result = await this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([
+        node('requestingUser'),
+        relation('out', '', 'pinned', { active: true }),
+        node('node', 'BaseNode', { id }),
+      ])
+      .return('node')
+      .first();
+    return result ? true : false;
+  }
+
+  async add(id: string, session: Session): Promise<void> {
+    const createdAt = DateTime.local();
+    await this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'BaseNode', { id })])
+      .create([
```

Comment:
I don't think we can use merge here like below
```
.merge([
        node('requestingUser'),
        relation('out', '', 'pinned', { createdAt }),
        node('node', 'BaseNode', { id }),
])
```
Actually, there is no node with only the `BaseNode` label with this `id`... so it will create a node with only `BaseNode` label with this id and create a relation with the requestingUser instead of the project node. So the project node will miss the `:pinned` relation with the user.